File: challenges/challenge-04/C04.ts
export function resolve() {
    const valids: number[] = [];
    
    for (let n = 11098; n < 98123; n++ ) {
        const nStr = n.toString();
        const match = nStr.match(/5/g);
        const ascCount = [...nStr.substring(1)].reduce((acc, curr, i) => +curr >= +nStr[i] ? acc + 1 : acc, 0)
        
        if (match && match.length >= 2 && ascCount === 4) {
            valids.push(n);
        }
    }

    return `${valids.length}-${valids[55]}`
}

File: challenges/challenge-04/C04.py
from functools import reduce

def resolve():
    valids = []
    n_range = map(lambda n: str(n), range(11098, 98123))
    for nStr in n_range:
        have_two_five = len(nStr.replace('5', '')) <= 3
        asc_count = reduce(lambda acc, i: acc + 1 if int(nStr[i]) >= int(nStr[i-1]) else acc, range(1, 5), 0)
        if have_two_five and asc_count == 4:
            valids.append(nStr)

    return f"{len(valids)}-{valids[55]}"
